### PR TITLE
Fix benches and add them to CI

### PR DESCRIPTION
**Pull request by tiziano88**
There is more room for improvement in sharing logic between benches, tests and xtask, but I'll leave that for a separate PR.

Ref #3931 #3738

### CODE REVIEW

File: oak_functions_launcher/benches/integration_benches.rs [hunk 1]
```diff
@@ -41,18 +47,45 @@ struct OakFunctionsTestConfig {
...
 
+    runtime.block_on(xtask::testing::run_step(xtask::launcher::build_stage0()));
+    runtime.block_on(xtask::testing::run_step(xtask::launcher::build_binary(
+        "build Oak Restricted Kernel binary",
+        xtask::launcher::OAK_RESTRICTED_KERNEL_BIN_DIR
+            .to_str()
+            .unwrap(),
+    )));
+
     let oak_functions_linux_fd_bin_path =
```
Comment (jul-sh) [inline oak_functions_launcher/benches/integration_benches.rs:64]: given that this is now enclave app, we should rename this var.

File: kokoro/presubmit.sh [hunk 2]
```diff
@@ -11,7 +11,7 @@ export RUST_BACKTRACE=1
 export RUST_LOG=debug
 
 ./scripts/docker_pull
-./scripts/docker_run cargo nextest run --hide-progress-bar
+./scripts/docker_run cargo nextest run --all-targets --hide-progress-bar
```
Comment (jul-sh) [inline kokoro/presubmit.sh:14]: I'm assuming this is what results in benches being run in CI? Seems suprising.

**Review by jul-sh — APPROVED** [review]
lgtm. I'm a big fan of the fact that this also moves the benches from the linux version to the KVM version.

File: oak_functions_launcher/benches/integration_benches.rs (hunk 1 above)
Comment (tiziano88) [reply oak_functions_launcher/benches/integration_benches.rs:64]: Good catch, done.

File: kokoro/presubmit.sh (hunk 2 above)
Comment (tiziano88) [reply kokoro/presubmit.sh:14]: Yeah I was confused too, I always assumed --all-targets means it tries all the possible --targets, instead it's really so it also runs benches (with a single iteration, AFAICT) and examples, which is what we want. Added a comment.

Comment (jul-sh) [reply kokoro/presubmit.sh:15]: Sweet, thx for adding the comment!

**Pull request merged**